### PR TITLE
IDE-3500 use gw watch way if support

### PR DESCRIPTION
**Pull request by jtydhr88**

### CODE REVIEW

**Comment by jtydhr88** [comment]
jenkins retest

**Comment by jtydhr88** [comment]
jenkins retest

File: tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleUtil.java [hunk 1]
```diff
@@ -96,7 +97,8 @@ public static IStatus importGradleProject(File dir, IProgressMonitor monitor) th
 
 		build.synchronize(NewProjectHandler.IMPORT_AND_MERGE, AsyncHandler.NO_OP);
 
-		waitImport();
+		// We should import asynchronous or not wait the gradle job finished
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleUtil.java]: This change should be another PR not dealing with Watch.  Can you create another ticket for this issue?

File: tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleProject.java [hunk 2]
```diff
@@ -354,4 +360,84 @@ private IFolder _createResorcesFolder(IProject project) {
...
+
+			@Override
+			public IStatus run(IProgressMonitor monitor) {
+				IStatus status = Status.OK_STATUS;
+
+				try {
+					GradleUtil.runGradleTask(getProject(), new String[] {
+						"watch"
+					}, new String[] {
+						"-t"
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleProject.java]: Lets use `--continuous` so we can easily see what is being done.

File: tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleProject.java [hunk 3]
```diff
@@ -354,4 +360,84 @@ private IFolder _createResorcesFolder(IProject project) {
...
+		if (ListUtil.isNotEmpty(jobs)) {
+			jobs[0].cancel();
+		}
+	}
+
+	@Override
+	public boolean supported() {
+		boolean supported = false;
+
+		ProjectConnection connection = null;
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleProject.java]: This is really slow.  Can we figure out some other way to determine if watch is supported?

File: tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/PortalServerLaunchConfigDelegate.java [hunk 4]
```diff
@@ -132,8 +136,30 @@ public void serverChanged( ServerEvent event )
                         LiferayServerCore.logError( "Could not reinitialize source lookup director", e );
                     }
                 }
+                else if((event.getKind() & ServerEvent.SERVER_CHANGE)>0 && event.getState() == IServer.STATE_STARTED) {
+					IModule[] modules = server.getModules();
+
+					for (IModule module : modules) {
+						IBundleProject project = LiferayCore.create(IBundleProject.class, module.getProject());
+						if (project != null && project instanceof IWatchableProject
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/PortalServerLaunchConfigDelegate.java]: So instead of having a `supported` just do this
```IWatchableProject watchableProject = LiferayCore.create(IWatchableProject.class)```
and then you don't need to do a `supported`

File: tools/plugins/com.liferay.ide.core/src/com/liferay/ide/core/IWatchableProject.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * @author Terry Jia
+ */
+public interface IWatchableProject {
+
+	public void watch();
+
+	public void unwatch();
+
+	public boolean supported();
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.core/src/com/liferay/ide/core/IWatchableProject.java]: We don't need `supported` see comment below.
